test/regression/riscv/instructions/float.cc

```diff
@@ -200,6 +200,30 @@ TEST_P(InstFloat, FDIV_S) {
   EXPECT_EQ(getFPRegister<float>(15), (float)999.212341);
   EXPECT_EQ(getFPRegister<float>(16), (float)999.212341 / (float)4.52432537);
   EXPECT_EQ(getFPRegister<float>(0), (float)999.212341 / (float)-3.78900003);
+
+  initialHeapData_.resize(32);
+  uint64_t* intHeap = reinterpret_cast<uint64_t*>(initialHeapData_.data());
+  intHeap[0] = 0xffffffff3f800000;  // Correct NaN boxing of no. 1
+  intHeap[1] = 0xfff7ffff3f800000;  // Incorrect NaN boxing of no. 1
+
+  RUN_RISCV(R"(
+    # Get heap address
+    li a7, 214
+    ecall
+
+    fld fa3, 0(a0) # Correct
+    fld fa4, 8(a0) # Incorrect
+
+    fdiv.s fa5, fa3, fa3  # 1 / 1 = 1
+    fdiv.s fa6, fa4, fa3  # Incorrect NaN box should be caught by fdiv and
+                          # canonical NaN used as input, NaN / 1 = NaN
+   )");
+
+  EXPECT_EQ(getFPRegister<uint64_t>(13), 0xffffffff3f800000);
+  EXPECT_EQ(getFPRegister<uint64_t>(14), 0xfff7ffff3f800000);
+
+  EXPECT_EQ(getFPRegister<uint64_t>(15), 0xffffffff3f800000);
+  EXPECT_EQ(getFPRegister<uint64_t>(16), 0xffffffff7fc00000);
```

Comment:
I presume `0xffffffff7fc00000` is the hex code for NAN?

Reply:
This is the hex code for a single precision NaN which has been NaN boxed to fit in a double precision register. Comment needed?

Reply:
Could be useful? Or use a static constexpr / define ?

Reply:
constexpr sounds good